### PR TITLE
Add activerecord 5.0 gemfile, update tests, fix deprecation warning

### PR DESCRIPTION
**Pull request by jordanwheeler**
I added an activerecord 5.0 gemfile, with the newer version of mysql2, which matches what we run in Shopify core.  All the tests now pass both here, and with the previous version of the Gemfile.  Nothing changed functionality wise, I basically just changed a bunch of stubbing stuff and simplified some of the tests as I went.  No actual code was touched.  But I didn't want to merge without at least opening a PR first 😄   You can try running the tests yourself if you want.

@sroysen @insom

### CODE REVIEW

File: spec/integration/chunker_spec.rb [hunk 1]
```diff
@@ -71,9 +71,9 @@ def printer.end(*) [] ;end
     it 'should throttle work stride based on slave lag' do
       5.times { |n| execute("insert into origin set id = '#{ (n * n) + 1 }'") }
 
-      printer = MiniTest::Mock.new
-      15.times { printer.expect(:notify, :return_value, [Fixnum, Fixnum]) }
-      printer.expect(:end, :return_value, [])
+      printer = mock()
+      printer.expects(:notify).with(instance_of(Fixnum), instance_of(Fixnum)).twice
```
Comment (insom) [inline spec/integration/chunker_spec.rb]: Why is this `.twice` when the old code was `15.times`?

(I mean, `.twice` looks correct though, with `max_retries` being two).

**Comment by insom** [comment]
```
Finished in 28.071124s, 1.9949 runs/s, 3.6336 assertions/s.

  1) Failure:
Lhm::changes#test_0013_should change the last column in a table [/Users/aaron/Repo/lhm/spec/integration/lhm_spec.rb:232]:
--- expected
+++ actual
@@ -1 +1 @@
-{:type=>"int(5)", :is_nullable=>"NO", :column_default=>"0"}
+{:type=>"int(5)", :is_nullable=>"NO", :column_default=>nil}
```

This may be my local environment, so if it works for you, 🚢 

New code is nicer and tidier too :+1:. We're getting Rails 5.1 deprecation warnings, but maybe LHM will be replaced by 👻  before Rails 5.1 ships?

File: spec/integration/chunker_spec.rb (hunk 1 above)
Comment (jordanwheeler) [inline spec/integration/chunker_spec.rb]: yeah i did a double take on that one too, this test is now correct.  i don't know where/when that broke, but that test does strides of 10, in two chunks, so yeah.  that's the correct behaviour.

**Comment by jordanwheeler** [comment]
those deprecation warnings are super annoying.  i almost changed the code like it asked but decided not to.  it's basically just telling us that the method we use now checks for the existence of tables and views, but in rails 5.1 that same method will only check for the existence of tables, and not views.  which is fine for us, so i didn't bother changing it.  doesn't really matter either way i don't think, i could just change the method and get rid of the spam.

**Comment by jordanwheeler** [comment]
alright I changed `table_exists?` to `data_source_exists?` like the deprecation warnings suggested.  again no functionality is changing, but I am touching actual code now 😉

**Comment by jordanwheeler** [comment]
Yeah that test failure seems weird @insom.  I just blew away my local instances and started from scratch and everything passes.  I wondered if maybe the default default value changed, but i'm definitely running these under activerecord 5.0.2 and I can't seem to replicate that.

**Comment by insom** [comment]
I thought it was `sql_mode` but even changing it to `NO_ENGINE_SUBSTITUTION` I get an error. I think it's harmless, especially if it's not even appearing for you. :+1:

**Pull request merged**